crates/polars-ops/src/chunked_array/list/namespace.rs

```diff
@@ -342,8 +342,12 @@ pub trait ListNameSpaceImpl: AsList {
     /// So index `0` would return the first item of every sublist
     /// and index `-1` would return the last item of every sublist
     /// if an index is out of bounds, it will return a `None`.
-    fn lst_get(&self, idx: i64) -> PolarsResult<Series> {
+    fn lst_get(&self, idx: i64, null_on_oob: bool) -> PolarsResult<Series> {
         let ca = self.as_list();
+        if !null_on_oob && ca.downcast_iter().any(|arr| index_is_oob(arr, idx)) {
+            polars_bail!(ComputeError: "get index is out of bounds");
+        }
```

Comment:
Could also use `lst_lengths` which would be cleaner but would collect all the indices into one container. I don't have a good sense of how much of an impact that'd really have on performance, though.

Reply:
We shouldn't use that. It's wasteful to collect a new container when we can iterate over the indices directly. We try to be a fast library. ;)